draft-ietf-ppm-dap.md

```diff
@@ -604,47 +726,60 @@ Clients upload reports by using an HTTP POST to `[leader]/upload`, where
 structured as follows:
 
 ~~~
+struct {
+    Time time;
+    Nonce nonce;
+    Extension extensions<0..2^16-1>;
+} ReportInfo;
+
 struct {
   TaskID task_id;
-  Nonce nonce;
-  Extension extensions<0..2^16-1>;
+  ReportInfo info;
   HpkeCiphertext encrypted_input_shares<1..2^32-1>;
 } Report;
 ~~~
 
-This message is called the client's report. It contains the following fields:
-
-* `task_id` is the task ID of the task for which the report is intended.
-* `nonce` is the report nonce generated by the client. This field is used by the
-  aggregators to ensure the report appears in at most one batch. (See
-  {{anti-replay}}.)
-* `extensions` is a list of extensions to be included in the Upload flow; see
-  {{upload-extensions}}.
-* `encrypted_input_shares` contains the encrypted input shares of each of the
-  aggregators. The order in which the encrypted input shares appear MUST match
-  the order of the task's `aggregator_endpoints` (i.e., the first share should
-  be the leader's, the second share should be for the first helper, and so on).
-
-To generate the report, the client begins by initializing the report nonce.
-Specifically, the client first sets `Nonce.rand` to 16 random bytes output from
-a cryptographically secure random number generator. It then sets `Nonce.time`
-to the number of seconds elapsed since the start of the UNIX epoch, rounded down to the nearest multiple of `min_batch_duration`.
-This truncation is done to ensure that the report's timestamp cannot be used to
-link the report back to the originating client.
-
-The client then finishes the report generation by sharding its measurement into
-a sequence of input shares as specified by the VDAF in use. To encrypt an input
-share, the client generates an {{HPKE}} ciphertext for the aggregator by running
+This message is called the Client's report. It consists of a "header" and the
+encrypted input share of each of the Aggregators. The header consists of:
+
+* The task ID for which the report is intended.
+
+* A timestamp representing the time at which the report was generated.
+  Specifically, the `time` field is set to the number of seconds elapsed since
+  the start of the UNIX epoch.
+
+    * For time-series tasks, the client MUST round this value to the nearest
```

Comment:
This wasn't a MUST previously, and we discussed why at the time in [#281](https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/pull/281#pullrequestreview-1025110801).

Reply:
Oh good, I thought the intention was MUST. How about SHOULD?